eng/pipelines/ci-typespec-java.yaml

```diff
@@ -38,6 +38,9 @@ jobs:
       vmImage: "MMSUbuntu20.04"
 
     steps:
+      - checkout: self
```

Comment:
Is the release process for typespec-java same as before?

Reply:
I hope so. (I think it is covered in this PR, but cannot be really sure till I trigger one)

I will test upon next release (probably on a branch that at first disables the publish).